src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/SettingsFragmentPresenter.kt

```diff
@@ -1086,6 +1087,33 @@ class SettingsFragmentPresenter(private val fragmentView: SettingsFragmentView)
                     R.string.use_black_backgrounds_description
                 )
             )
+
+            val deviceOrientationSetting: AbstractIntSetting = object : AbstractIntSetting {
+                override var int: Int
+                    get() = preferences.getInt(key, defaultValue)
+                    set(value) {
+                        preferences.edit()
+                            .putInt(key, value)
+                            .apply()
+                        settingsActivity.recreate()
+                    }
+                override val key: String = Settings.PREF_DEFAULT_SCREEN_ORIENTATION
+                override val section: String? = null
+                override val isRuntimeEditable: Boolean = true
+                override val valueAsString: String
+                    get() = int.toString()
+                override val defaultValue: Int = ActivityInfo.SCREEN_ORIENTATION_USER
+            }
+
+            add(
+                SingleChoiceSetting(
+                    deviceOrientationSetting,
+                    R.string.device_orientation_title,
+                    R.string.device_orientation_description,
+                    R.array.deviceOrientationEntries,
+                    R.array.deviceOrientationValues
+                )
+            )
```

Comment:
This setting really shouldn't be in the theme options, as it has nothing to do with themes and also makes the setting impossible to change while in-game. I would instead recommend placing it in the in-game sidebar, which is more relevant to the functionality and is more accessible while in-game, which is likely when you'd want to be changing the setting.

Reply:
I initially just put it there because i wasn't sure where else.

I don't think it should be in the side-bar, there is enough there already. I was thinking about putting it in the "graphics" section, which doesn't _quite_ fit, but is also not that far off.

Dolphin and others for example put this setting in the "interface" section, which is roughly equivalent to "theming" here.

> which is likely when you'd want to be changing the setting.

The idea here is that we make it configurable enough that you _don't_ have to change it after the first time.

Reply:
> The idea here is that we make it configurable enough that you don't have to change it after the first time.

It's not unreasonable to assume that there will be people who load up a game, decide they want to change their screen orientation for whatever reason, and then have to exit out of the game entirely to change it.

Reply:
> It's not unreasonable to assume that there will be people who load up a game, decide they want to change their screen orientation for whatever reason, and then have to exit out of the game entirely to change it.

That is the reason why i want to keep the auto-rotate setting. This would solve that issue. (Except for the even-rarer case where the user then also wants to re-lock the screen into a certain position, but to be honest, this would be a far strech anyway, since that means that the content of the screen is rotated in relation to the user)

The value `SCREEN_ORIENTATION_SENSOR` allows the device to be rotated in any direction, except inverse-portrait. It blocks the device from rotating "freely" so that even lying down the device will still be in landscape. (This is how many video apps do it, when not beeing fixed by other settings)